### PR TITLE
Refactor UserData.displayName

### PR DESCRIPTION
**Pull request by makingthematrix**
From time to time a bug comes up in our UI, usually about that the user name does not appear correctly in one place or another. After investigation, we find out that we used `UserData.name` or
`UserData.displayName` instead of `UserData.getDisplayName`.
This PR removes both`UserData.displayName` and  `UserData.getDisplayName`, and adds a migration to remove the column from the Users table in DB as well. From now on we can safely use `name` in all cases.
#### APK
[Download build #786](http://10.10.124.11:8080/job/Pull%20Request%20Builder/786/artifact/build/artifact/wire-dev-PR2523-786.apk)
[Download build #794](http://10.10.124.11:8080/job/Pull%20Request%20Builder/794/artifact/build/artifact/wire-dev-PR2523-794.apk)

### CODE REVIEW

**Comment by makingthematrix** [comment]
build again

**Comment by makingthematrix** [comment]
@android10 
The original code was there for a long time as a sort of a preparation for a feature where users can change names of their contacts. So `name` would be the original name of the contact, and `displayName` would be the name made up by the user. Then, `getDisplayName()` would be used to get one or the other (`displayName`, or if it's not set, then `name`).
But the functionality was never finished and tbh it was never clear which one of the three should be used when. Every few months we got a bug that the name was not displayed or not updated in some circumstances, because the wrong field was used.

But this is no longer important. As I worked on this code, I realized I can simply remove `displayName` altogether. In current code it just carries the copy of `name`.

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/db/ZMessagingDB.scala [hunk 1]
```diff
@@ -390,6 +390,49 @@ object ZMessagingDB {
...
+        s"""
+           | UPDATE Users
+           | SET name = display_name
+           | WHERE display_name IS NOT NULL AND display_name IS NOT "" AND name IS NOT display_name
+        """.stripMargin)
+      db.execSQL(
+        """
+          | CREATE TABLE UsersCopy (
+          | _id TEXT PRIMARY KEY,
+          | teamId TEXT, name TEXT, email TEXT, phone TEXT, tracking_id TEXT,
```
Comment (gizemb) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/db/ZMessagingDB.scala]: i guess you dropped the `displayName` and unified them under `name` property, right?

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/content/UsersStorage.scala [hunk 2]
```diff
@@ -117,27 +117,27 @@ class UsersStorageImpl(context: Context, storage: ZmsDatabase)
...
     }
-  }, "UsersDisplayNameUpdater")
+  }, "UsersNameUpdater")
 
-  def updateDisplayNamesWithSameFirst(users: Seq[UserId], cs: mutable.HashMap[UserId, NameParts]): Unit = {
-    def setFullName(user: UserId) = update(user, { (u : UserData) => u.copy(displayName = u.name) })
-    def setDisplayName(user: UserId, name: String) = update(user, (_: UserData).copy(displayName = Name(name)))
+  def updateNamesWithSameFirst(users: Seq[UserId], cs: mutable.HashMap[UserId, NameParts]): Unit = {
+    def setFullName(user: UserId) = update(user, { (u : UserData) => u.copy(name = u.name) })
+    def setDisplayName(user: UserId, name: String) = update(user, (_: UserData).copy(name = Name(name)))
```
Comment (gizemb) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/content/UsersStorage.scala]: what's the difference between setFullName and setDisplayName now?

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/log/LogShowInstancesSE.scala [hunk 3]
```diff
@@ -146,7 +146,7 @@ trait LogShowInstancesSE {
   implicit val UserDataLogShow: LogShow[UserData] =
     LogShow.createFrom { u =>
       import u._
-      l"UserData(id: $id | teamId: $teamId | name: $name | displayName: $displayName | email: $email | phone: $phone | handle: $handle | deleted: $deleted)"
+      l"UserData(id: $id | teamId: $teamId | name: $name | displayName: $name | email: $email | phone: $phone | handle: $handle | deleted: $deleted)"
```
Comment (gizemb) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/log/LogShowInstancesSE.scala]: we can omit displayName now

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/db/ZMessagingDB.scala (hunk 1 above)
Comment (gizemb) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/db/ZMessagingDB.scala]: i saw your comment now. marking this as resolved.

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/log/LogShowInstancesSE.scala (hunk 3 above)
Comment (makingthematrix) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/log/LogShowInstancesSE.scala]: done

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/content/UsersStorage.scala (hunk 2 above)
Comment (makingthematrix) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/content/UsersStorage.scala]: code removed

**Pull request merged**